pkg/datapath/loader/loader.go

```diff
@@ -790,16 +777,31 @@ func replaceWireguardDatapath(ctx context.Context, logger *slog.Logger, lnc *dat
 			)
 		}
 	}
-	// Selectively detach cil_from_netdev from ingress.
-	if !option.Config.NeedBPFHostOnWireGuardDevice() {
-		if err := detachSKBProgram(logger, device, symbolFromHostNetdevEp,
+	// Attach/detach cil_from_wireguard to/from ingress.
+	if option.Config.NeedIngressOnWireGuardDevice() {
+		if err := attachSKBProgram(logger, device, obj.FromWireguard, symbolFromWireguard,
+			linkDir, netlink.HANDLE_MIN_INGRESS, option.Config.EnableTCX); err != nil {
+			return fmt.Errorf("interface %s ingress: %w", device, err)
+		}
+	} else {
+		if err := detachSKBProgram(logger, device, symbolFromWireguard,
 			linkDir, netlink.HANDLE_MIN_INGRESS); err != nil {
 			logger.Error("",
 				logfields.Error, err,
 				logfields.Device, wgTypes.IfaceName,
 			)
 		}
 	}
+	// Cleanup previous cil_from_netdev from v1.17.
+	if err := detachSKBProgram(logger, device, symbolFromHostNetdevEp,
+		linkDir, netlink.HANDLE_MIN_INGRESS); err != nil {
+		logger.Error("",
+			logfields.Error, err,
+			logfields.Device, wgTypes.IfaceName,
+		)
+	}
+	// Cleanup previous calls map from v1.17.
+	os.RemoveAll(filepath.Join(bpf.TCGlobalsPath(), fmt.Sprintf("cilium_calls_wireguard_%d", identity.ReservedIdentityWorld)))
```

Comment:
nit: might be best to emit errors that aren't "file not found".

Reply:
Since removing such map is not a functional problem, I wouldn't emit errors at this point during the datapath regeneration. Maybe a warning/info. WDYT?

Reply:
yeah warning makes sense

Reply:
I'm not fully convinced yet, will send a future patch in case where I update all the `cleanCallsMaps()` usages 😞 
The error seems not to be checked in most of the `os.Remove` usages, and I'm not sure a warn/info message here  would improve debuggability (the map is not in use anyway).